lib/mayaUsd/ufe/UsdObject3d.cpp

```diff
@@ -99,19 +80,23 @@ Ufe::BBox3d UsdObject3d::boundingBox() const
 
 bool UsdObject3d::visibility() const
 {
-    auto visAttr = getVisibilityAttribute(sceneItem());
-    if (visAttr) {
-        return (visAttr->get() != UsdGeomTokens->invisible);
+    TfToken visibilityToken;
+    auto    visAttr = UsdGeomImageable(fPrim).GetVisibilityAttr();
+    visAttr.Get(&visibilityToken);
+
+    bool result;
```

Comment:
Your code below is structured such that there might be token values other than invisible or inherited; if that's the case (and I don't believe it is possible), result would be uninitialized.  For safety and conciseness, I would go with something like the existing code, i.e.
return visibilityToken != UsdGeomTokens->invisible;

Reply:
Yeah, I thought about this but couldn't convince myself that this ever change. I mean what else could be added? :)
https://graphics.pixar.com/usd/docs/api/class_usd_geom_imageable.html#a64bae16002f3448a6cdeee55e8ebf1fc
I am happy to change it, would be less branching as well.

Reply:
Please see 696db13a64c0795142b7035d0d78d95fc0607782